### PR TITLE
Issue284: Product voucher reservation

### PR DESCRIPTION
**Pull request by jknnr**
teamforus/me#284

### CODE REVIEW

File: data/src/main/java/io/forus/me/android/data/net/vouchers/VouchersService.kt [hunk 1]
```diff
@@ -10,19 +10,32 @@ import retrofit2.http.Path
 
 interface VouchersService {
 
+
+    /*
+ budget:
+{{apiUrl}}/platform/provider/vouchers/0xc87fbc0f088057b25b3831c7fc905bc36744f462
+product_vouchers:
```
Comment (maxvisser) [inline data/src/main/java/io/forus/me/android/data/net/vouchers/VouchersService.kt]: is this needed?

File: presentation/build.gradle [hunk 2]
```diff
@@ -116,7 +117,7 @@ android {
         dev {
             dimension "version"
             versionNameSuffix "-dev"
-            buildConfigField "String", SERVER_URL, "\"https://dev.api.forus.io/\""
+            buildConfigField "String", SERVER_URL, "\"https://test.api.forus.rminds.dev/\"" //\"https://dev.api.forus.io/\""
```
Comment (maxvisser) [inline presentation/build.gradle]: which will it use?

File: presentation/src/main/java/io/forus/me/android/presentation/view/screens/account/assigndelegates/email/RestoreByEmailFragment.kt [hunk 3]
```diff
@@ -129,8 +129,9 @@ class RestoreByEmailFragment : ToolbarLRFragment<RestoreByEmailModel, RestoreByE
         email.isEditable = !(vs.model.sendingRestoreByEmailSuccess == true)
 
         if(vs.model.sendingRestoreByEmailSuccess == true && !instructionsAlreadyShown){
-            InstructionsDialog(context!!).show()
-            instructionsAlreadyShown = true
+           /* InstructionsDialog(context!!).show()
```
Comment (maxvisser) [inline presentation/src/main/java/io/forus/me/android/presentation/view/screens/account/assigndelegates/email/RestoreByEmailFragment.kt]: needs to be cleaned up

File: presentation/src/main/java/io/forus/me/android/presentation/view/screens/account/newaccount/NewAccountFragment.kt [hunk 4]
```diff
@@ -118,6 +119,8 @@ class NewAccountFragment : ToolbarLRFragment<NewAccountModel, NewAccountView, Ne
     override fun render(vs: LRViewState<NewAccountModel>) {
         super.render(vs)
 
+        Log.d("testforus","NewAccountRender")
```
Comment (maxvisser) [inline presentation/src/main/java/io/forus/me/android/presentation/view/screens/account/newaccount/NewAccountFragment.kt]: Log.d can also go

File: presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt [hunk 5]
```diff
@@ -89,6 +89,16 @@ class QrActionProcessor(private val scanner: QrScannerActivity,
                     .show()
     }
 
+    /*
```
Comment (maxvisser) [inline presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt]: cleanup

File: presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt [hunk 6]
```diff
@@ -155,13 +167,50 @@ class QrActionProcessor(private val scanner: QrScannerActivity,
             }, 1000)
         } else {
             showToastMessage(resources.getString(R.string.qr_voucher_scanned))
-            navigator.navigateToVoucherProvider(scanner, address)
+            scanHasProductVouchers(address,isAvialableScannedVoucher)
+            //navigator.navigateToProductReservation(scanner, address)
```
Comment (maxvisser) [inline presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt]: cleanup

File: presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt [hunk 7]
```diff
@@ -155,13 +167,50 @@ class QrActionProcessor(private val scanner: QrScannerActivity,
...
             (android.os.Handler()).postDelayed({
                 reactivateDecoding()
             }, 1000)
         }
     }
 
+
+    private fun scanHasProductVouchers(address: String, isAvialableScannedVoucher: Boolean) {
+        Log.d("testforus","scanHasProductVouchers 0")
```
Comment (maxvisser) [inline presentation/src/main/java/io/forus/me/android/presentation/view/screens/qr/QrActionProcessor.kt]: log.d cleanup

**Comment by maxvisser** [comment]
- [ ] cleanup debug / comments
- [ ] fix returning to reservation list

File: data/src/main/java/io/forus/me/android/data/net/vouchers/VouchersService.kt (hunk 1 above)
Comment (maestrovsmaster) [reply data/src/main/java/io/forus/me/android/data/net/vouchers/VouchersService.kt]: No, i will delete

File: presentation/build.gradle (hunk 2 above)
Comment (maestrovsmaster) [reply presentation/build.gradle]: For dev it will use rminds  temporary domain for dev:  https://test.api.forus.rminds.dev
If needed, i can remove it

**Pull request merged**